src/mongo/db/repl/replica_set_config.cpp

```diff
@@ -574,7 +587,7 @@ const int ReplicaSetConfig::findMemberIndexByHostAndPort(const HostAndPort& hap)
     int x = 0;
     for (std::vector<MemberConfig>::const_iterator it = _members.begin(); it != _members.end();
          ++it) {
-        if (it->getHostAndPort() == hap) {
+        if (it->getHostInternalAndPort() == hap || it->getHostAndPort() == hap) {
```

Comment:
I'm not sure this is correct. I believe this should only use the internal HostAndPort, but we should check the callers to confirm.